### PR TITLE
Admin word: index - show - create - update - delete

### PR DESCRIPTION
**Pull request by truongnv0211**

### CODE REVIEW

File: app/views/admin/words/_word.html.erb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<tr>
+  <td><%= word.jp_word %></td>
+  <td><%= word.answer_1%></td>
```
Comment (NguyenTanDuc) [inline app/views/admin/words/_word.html.erb]: space %>

File: app/views/admin/words/show.html.erb [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    <th><%= t :answer_4 %></th>
+    <td>: <%= @word.answer_4 %></td>
+  </tr>
+  <tr>
+    <th><%= t :correct_answer %></th>
+    <td>: <%= @word.correct_answer %></td>
+  </tr>
+  <tr>
+    <th><%= t :category %></th>
+    <td>: <%= Category.find(@word.category_id).title%></td>
```
Comment (NguyenTanDuc) [inline app/views/admin/words/show.html.erb]: space %>

File: app/controllers/admin/words_controller.rb [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    redirect_to admin_words_path
+  end
+
+  private
+  def set_word
+    @word = Word.find params[:id]
+  end
+
+  def word_params
+    params.require(:word).permit :jp_word, :answer_1, :answer_2, :answer_3,
```
Comment (phamkykhoi) [inline app/controllers/admin/words_controller.rb]: Dùng nested attribute

File: config/locales/en.yml [hunk 4]
```diff
@@ -47,8 +47,24 @@ en:
   delete: Delete
   confirm_delete: "You sure?"
   user_delete_message: "User deleted!"
+<<<<<<< HEAD
```
Comment (phamhuycuong) [inline config/locales/en.yml]: rebase?

File: app/views/admin/words/_answer_fields.html.erb [hunk 5]
```diff
@@ -0,0 +1,18 @@
+<fieldset>
+  <table>
+    <thead>
+      <tr>
+        <th><%= f.label :answer_content, "Answer" %></th>
```
Comment (khanhkhanh) [inline app/views/admin/words/_answer_fields.html.erb]: i18n

File: app/views/admin/words/_word.html.erb [hunk 6]
```diff
@@ -0,0 +1,7 @@
+<tr>
+  <td><%= word.jp_word %></td>
+  <td><%= Category.find(word.category_id).title  %></td>
+  <td><%= link_to t("show"), admin_word_path(word) %></td>
+  <td><%= link_to t("edit"), edit_admin_word_path(word) %></td>
+  <td><%= link_to t("destroy") , admin_word_path(word), method: :delete, data: {confirm: t("confirm_delete")} %></td>
```
Comment (khanhkhanh) [inline app/views/admin/words/_word.html.erb]: Remove redundant whitespace

File: app/controllers/admin/words_controller.rb [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+  end
+
+  private
+  def set_word
+    @word = Word.find params[:id]
+  end
+
+  def word_params
+    params.require(:word).permit :jp_word, :category_id,
+                                  answers_attributes: [:id, :answer_content, :correct,:_destroy]
```
Comment (khanhkhanh) [inline app/controllers/admin/words_controller.rb]: Indent

File: app/helpers/application_helper.rb [hunk 8]
```diff
@@ -7,4 +7,18 @@ def full_title page_title = ""
...
+    f.hidden_field(:_destroy) + link_to(name, "", onclick: "remove_fields(this); return false")
+  end
+
+  def link_to_add_fields(name, f, association, cssClass, title)
+    new_object = f.object.class.reflect_on_association(association).klass.new
+    fields = f.fields_for(association, new_object, child_index: "new_#{association}") do |builder|
+      render(association.to_s.singularize + "_fields", f: builder)
+    end
+    link_to name, "#", onclick: h("add_fields(this, \"#{association}\", \"#{escape_javascript(fields)}\")"),
+            class: cssClass, title: title, remote: true
```
Comment (khanhkhanh) [inline app/helpers/application_helper.rb]: Indent

File: app/models/word.rb [hunk 9]
```diff
@@ -1,9 +1,12 @@
 class Word < ActiveRecord::Base
   belongs_to :category
   has_many :lesson_words, dependent: :destroy
+  has_many :answers, dependent: :destroy
+  accepts_nested_attributes_for :answers, allow_destroy: true,
+                                 reject_if: lambda {|a| a[:answer_content].blank? }
```
Comment (khanhkhanh) [inline app/models/word.rb]: Indent

File: app/views/admin/words/_form.html.erb [hunk 10]
```diff
@@ -0,0 +1,15 @@
+<%= form_for [:admin, @word] do |f| %>
+  <%= f.label :jp_word %>
+  <%= f.text_field :jp_word, class: "form-control" %>
+
+  <%= f.fields_for :answers do |builder| %>
+    <%= render "answer_fields", f: builder %>
+  <% end %>
+
+  <%= link_to_add_fields t("add_answer"), f, :answers, "btn btn-sm btn-primary", t("add_new_answer") %><br /><br />
```
Comment (khanhkhanh) [inline app/views/admin/words/_form.html.erb]: Lack of class attribute

Comment (khanhkhanh) [inline app/views/admin/words/_form.html.erb]: Lack of class attribute

Comment (khanhkhanh) [inline app/views/admin/words/_form.html.erb]: t("add_new_answer") is redundant?

File: app/views/admin/words/_form.html.erb [hunk 11]
```diff
@@ -0,0 +1,15 @@
...
+
+  <%= f.label :category %>
+  <%= f.collection_select(:category_id, Category.all, :id , :title, {prompt: true}, {class: "form-control"}) %>
```
Comment (khanhkhanh) [inline app/views/admin/words/_form.html.erb]: Move Category.all into controller

Comment (khanhkhanh) [inline app/views/admin/words/_form.html.erb]: Remove the outermost ()

File: app/views/admin/words/show.html.erb [hunk 12]
```diff
@@ -0,0 +1,24 @@
+<table class="table-striped table-condensed">
+  <tr>
+    <th><%= t :jp_word %></th>
+    <td>: <%= @word.jp_word %></td>
+  </tr>
+  <% @word.answers.each_with_index do |answer, index| %>
+    <tr>
+      <th><%= t "answer_#{index+1}" %></th>
+        <% if answer.correct %>
+      <td>: <%= answer.answer_content %> - <%= t :correct %></td>
```
Comment (khanhkhanh) [inline app/views/admin/words/show.html.erb]: ??? Indent ???

File: app/views/admin/words/show.html.erb [hunk 13]
```diff
@@ -0,0 +1,24 @@
+<table class="table-striped table-condensed">
+  <tr>
+    <th><%= t :jp_word %></th>
+    <td>: <%= @word.jp_word %></td>
+  </tr>
+  <% @word.answers.each_with_index do |answer, index| %>
+    <tr>
+      <th><%= t "answer_#{index+1}" %></th>
+        <% if answer.correct %>
```
Comment (khanhkhanh) [inline app/views/admin/words/show.html.erb]: ??? Terrible indentation ???

File: app/models/word.rb [hunk 14]
```diff
@@ -1,9 +1,12 @@
 class Word < ActiveRecord::Base
   belongs_to :category
   has_many :lesson_words, dependent: :destroy
+  has_many :answers, dependent: :destroy
+  accepts_nested_attributes_for :answers, allow_destroy: true,
+                                 reject_if: lambda {|attribute| attribute[:answer_content].blank? }
```
Comment (khanhkhanh) [inline app/models/word.rb]: Redundant space

File: app/views/admin/words/_form.html.erb [hunk 15]
```diff
@@ -0,0 +1,14 @@
...
+  <%= f.label :jp_word %>
+  <%= f.text_field :jp_word, class: "form-control" %>
+
+  <%= f.fields_for :answers do |builder| %>
+    <%= render "answer_fields", f: builder %>
+  <% end %>
+  <%= link_to_add_fields t("add_answer"), f, :answers, "btn btn-sm btn-primary", t("add_new_answer") %><br /><br />
+
+  <%= f.label :category %>
+  <%= f.collection_select(:category_id, Category.all, :id , :title, {prompt: true}, {class: "form-control"}) %>
```
Comment (khanhkhanh) [inline app/views/admin/words/_form.html.erb]: Move Category.all into controller and remove the outermost ()

File: app/views/admin/words/_form.html.erb [hunk 16]
```diff
@@ -0,0 +1,14 @@
+<%= form_for [:admin, @word] do |f| %>
+  <%= f.label :jp_word %>
+  <%= f.text_field :jp_word, class: "form-control" %>
+
+  <%= f.fields_for :answers do |builder| %>
+    <%= render "answer_fields", f: builder %>
+  <% end %>
+  <%= link_to_add_fields t("add_answer"), f, :answers, "btn btn-sm btn-primary", t("add_new_answer") %><br /><br />
```
Comment (khanhkhanh) [inline app/views/admin/words/_form.html.erb]: t("add_new_answer") is redundant?

File: app/views/admin/words/show.html.erb [hunk 17]
```diff
@@ -0,0 +1,24 @@
...
+      <td>: <%= answer.answer_content %> - <%= t :correct %></td>
+      <% else %>
+        <td>: <%= answer.answer_content %></td>
+      <% end %>
+    </tr>
+  <% end %>
+  </tr>
+  <tr>
+    <th><%= t :category %></th>
+    <td>: <%= Category.find(@word.category_id).title %></td>
```
Comment (khanhkhanh) [inline app/views/admin/words/show.html.erb]: @word.category.title

File: app/views/admin/words/_form.html.erb [hunk 18]
```diff
@@ -0,0 +1,14 @@
...
+  <%= f.label :jp_word %>
+  <%= f.text_field :jp_word, class: "form-control" %>
+
+  <%= f.fields_for :answers do |builder| %>
+    <%= render "answer_fields", f: builder %>
+  <% end %>
+  <%= link_to_add_fields t("add_answer"), f, :answers, "btn btn-sm btn-primary" %><br /><br />
+
+  <%= f.label :category %>
+  <%= f.collection_select :category_id, @categories, :id , :title, {prompt: true}, {class: "form-control"} %>
```
Comment (khanhkhanh) [inline app/views/admin/words/_form.html.erb]: Try removing all {}

File: app/views/admin/words/show.html.erb [hunk 19]
```diff
@@ -0,0 +1,24 @@
+<table class="table-striped table-condensed">
+  <tr>
+    <th><%= t :jp_word %></th>
+    <td>: <%= @word.jp_word %></td>
+  </tr>
+  <% @word.answers.each_with_index do |answer, index| %>
+  <tr>
```
Comment (khanhkhanh) [inline app/views/admin/words/show.html.erb]: Indentation

File: app/controllers/admin/words_controller.rb [hunk 20]
```diff
@@ -0,0 +1,57 @@
...
+  end
+
+  private
+  def set_word
+    @word = Word.find params[:id]
+  end
+
+  def word_params
+    params.require(:word).permit :jp_word, :category_id,
+      answers_attributes: [:id, :answer_content, :correct,:_destroy]
```
Comment (khanhkhanh) [inline app/controllers/admin/words_controller.rb]: Whitespace

File: app/views/admin/words/show.html.erb [hunk 21]
```diff
@@ -0,0 +1,24 @@
+<table class="table-striped table-condensed">
+  <tr>
+    <th><%= t :jp_word %></th>
+    <td>: <%= @word.jp_word %></td>
+  </tr>
+  <% @word.answers.each_with_index do |answer, index| %>
+    <tr>
+      <th><%= t "answer_#{index+1}" %></th>
+      <% if answer.correct %>
+      <td>: <%= answer.answer_content %> - <%= t :correct %></td>
```
Comment (khanhkhanh) [inline app/views/admin/words/show.html.erb]: ????

Comment (truongnv0211) [inline app/views/admin/words/show.html.erb]: show correct answer

Comment (NguyenTanDuc) [inline app/views/admin/words/show.html.erb]: indent

File: app/views/admin/words/show.html.erb [hunk 22]
```diff
@@ -0,0 +1,24 @@
+<table class="table-striped table-condensed">
+  <tr>
+    <th><%= t :jp_word %></th>
+    <td>: <%= @word.jp_word %></td>
+  </tr>
+  <% @word.answers.each_with_index do |answer, index| %>
+    <tr>
+      <th><%= t "answer_#{index+1}" %></th>
+      <% if answer.correct %>
+      <td>: <%= answer.answer_content %>-<%= t :correct %></td>
```
Comment (khanhkhanh) [inline app/views/admin/words/show.html.erb]: ??>???????????????

File: app/controllers/admin/words_controller.rb [hunk 23]
```diff
@@ -0,0 +1,57 @@
...
+  end
+
+  def new
+    @categories = Category.all
+    @word = Word.new
+    @word.answers.build
+  end
+
+  def create
+    @categories = Category.all
```
Comment (phamkykhoi) [inline app/controllers/admin/words_controller.rb]: Remove this line

File: app/views/admin/words/show.html.erb (hunk 22 above)
Comment (phamkykhoi) [inline app/views/admin/words/show.html.erb]: Indent

File: app/views/admin/words/show.html.erb [hunk 24]
```diff
@@ -0,0 +1,24 @@
...
+      <% else %>
+      <td>: <%= answer.answer_content %></td>
```
Comment (phamkykhoi) [inline app/views/admin/words/show.html.erb]: Indent

File: app/views/admin/words/_word.html.erb [hunk 25]
```diff
@@ -0,0 +1,7 @@
+<tr>
+  <td><%= word.jp_word %></td>
+  <td><%= word.category.title %></td>
+  <td><%= link_to t("show"), admin_word_path(word) %></td>
+  <td><%= link_to t("edit"), edit_admin_word_path(word) %></td>
+  <td><%= link_to t("destroy"), admin_word_path(word), method: :delete, data: {confirm: t("confirm_delete")} %></td>
```
Comment (vjnhat) [inline app/views/admin/words/_word.html.erb]: > 80 characters

File: app/controllers/admin/words_controller.rb (hunk 23 above)
Comment (truongnv0211) [inline app/controllers/admin/words_controller.rb]: I cant remove it. It use validate category_id, if remove it i cant submit form

File: app/views/admin/words/new.html.erb [hunk 26]
```diff
@@ -0,0 +1,8 @@
+<% provide :title, t("new_word") %>
```
Comment (nguyenngoctruong) [inline app/views/admin/words/new.html.erb]: remove ()

File: app/views/admin/words/edit.html.erb [hunk 27]
```diff
@@ -0,0 +1,7 @@
+<% provide :title, t("edit_word") %>
```
Comment (nguyenngoctruong) [inline app/views/admin/words/edit.html.erb]: remove ()

File: app/views/admin/words/_form.html.erb [hunk 28]
```diff
@@ -0,0 +1,15 @@
...
+  <%= f.text_field :jp_word, class: "form-control", required: true %>
+
+  <%= f.fields_for :answers do |builder| %>
+    <%= render "answer_fields", f: builder %>
+  <% end %>
+  <%= link_to_add_fields t("add_answer"), f, :answers, "btn btn-sm btn-primary" %><br /><br />
+
+  <%= f.label :category %>
+  <%= f.collection_select :category_id, @categories, :id , :title,
+  {prompt: true, selected: :first}, class: "form-control" %>
```
Comment (phamkykhoi) [inline app/views/admin/words/_form.html.erb]: indent

**Comment by dieunb** [comment]
OK

**Pull request merged**